src/Symfony/Component/HttpFoundation/Session/Attribute/NamespacedAttributeBag.php

```diff
@@ -124,7 +124,13 @@ protected function &resolveAttributePath($name, $writeContext = false)
 
         foreach ($parts as $part) {
             if (null !== $array && !array_key_exists($part, $array)) {
-                $array[$part] = $writeContext ? array() : null;
+                if (!$writeContext) {
+                    $null = null;
+
+                    return $null;
```

Comment:
You can do `return null;` instead

Reply:
The trick is that function returns by reference `protected function &resolveAttributePath($name, $writeContext = false)`.

Reply:
You're returning `$null` here, which does _not_ exist before the assignment here, so it's not referencing to anything

Reply:
Exactly, in php you cannot return immediate values when funciton returns reference:

```php
function &f() {
    return 'value'; // Notice:  Only variable references should be returned by reference
}
```

So, a trick is required:

```php
function &f() {
    $dummy = 'value';

    return $dummy; // Fine
}
```

Anyway when we need to return null we can shorten the hack to:

```php
function &f() {
    return $null; // Fine
}
```
because the result is exactly `null`:
```php
true === is_null(f());
```

Cheers